src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private Object[] keys = new Object[STORAGE_SIZE];
+    private Object[] values = new Object[STORAGE_SIZE];
+    private int size;
+    private int index;
+
+    public StorageImpl() {
+
+    }
+
+    public StorageImpl(Object[] keys, Object[] values, int size) {
+        this.keys = keys;
+        this.values = values;
+        this.size = size;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (hasAKey(key)) {
+            values[index] = value;
+            return;
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
```

Comment:
And then use it here:
```suggestion
    public void put(K key, V value) {
        int index = getKeyIndex(key);
        if (index != -1) {
            values[index] = value;
            return;
        }
        keys[size] = key;
        values[size] = value;
        size++;
    }
```
And similarly in other method